pulser-core/pulser/sequence/sequence.py

```diff
@@ -364,6 +365,132 @@ def config_slm_mask(self, qubits: Iterable[QubitId]) -> None:
+                )
+                if not (basis_match and addressing_match):
+                    channel_match[o_d_ch_name] = None
+                    channel_match[
+                        "ch_type_er_mess"
+                    ] = f"No match for channel {o_d_ch_name}."
+                    continue
+                if not strict:
+                    channel_match[o_d_ch_name] = n_d_ch_id
+                    break
+                ch_samples = sample_seq.channel_samples[o_d_ch_name]
+                ch_sample_phase = ch_samples.phase
+                # Find if there is phase change between pulses or not
+                phase_is_constant = True
+                if ch_sample_phase.size != 0:
+                    phase_is_constant = bool(
+                        np.all(ch_sample_phase == ch_sample_phase[0])
+                    )
+                # Phase_jump_time and cloc_period check
+                phase_jump_time_check = (
+                    o_d_ch_obj.phase_jump_time == n_d_ch_obj.phase_jump_time
+                )
+                clock_period_check = (
+                    o_d_ch_obj.clock_period % n_d_ch_obj.clock_period == 0
+                )
+                if phase_is_constant:
+                    if clock_period_check:
+                        channel_match[o_d_ch_name] = n_d_ch_id
+                        channel_match["alert_phase_jump"] = (
+                            phase_jump_time_check is False
+                        )
+                        break
+                    else:
+                        channel_match[o_d_ch_name] = None
+                        channel_match["strict_error_message"] = (
+                            "No channel with phase_jump_time"
+                            + " & clock_period match."
+                        )
+                    continue
+                if phase_jump_time_check and clock_period_check:
+                    channel_match[o_d_ch_name] = n_d_ch_id
+                    break
+                else:
+                    channel_match[o_d_ch_name] = None
+                    channel_match["strict_error_message"] = (
+                        "No channel with phase_jump_time"
+                        + " & clock_period match."
+                    )
+
+        if None in channel_match.values():
+            if channel_match["strict_error_message"] != "":
+                raise ValueError(channel_match["strict_error_message"])
+            else:
+                raise TypeError(channel_match["ch_type_er_mess"])
+        # Initialize the new sequence
+        new_seq = Sequence(self.register, new_device)
+
+        for call in self._calls[1:]:
+            if not (call.name == "declare_channel"):
+                getattr(new_seq, call.name)(*call.args, **call.kwargs)
+                continue
+            sw_channel_args = list(call.args)
+            # Switch the old id with the correct id
+            sw_channel_args[1] = channel_match[sw_channel_args[0]]
```

Comment:
Here you also have to account for the possibility that the channel ID was given as a keyword arg, so it won't be under `call.args` . Imagine the user declares the channel like this:
```python
seq.declare_channel("ch1", channel_id="rydberg_global")
```
Add an example like this to your tests and you'll see it will fail.
You can see an example on how this can be handled here: https://github.com/pasqal-io/Pulser/blob/1984e378069a1de536016aecf71195cd4c29f3e3/pulser-core/pulser/json/abstract_repr/serializer.py#L51-L67

Reply:
When you call a python function `def f(a, b, c=0): ...` you can give positional arguments (ie args) as keyword arguments (e.g. `f(1, b=2, c=3)` is valid). 
If this happens in declare channel (e.g. `seq.declare_channel("ch1", channel_id="rydberg_global"`), the store _Call will have `args=("ch1",)` and `kwargs={"channel_id": "rydberg_global"}`. 
In contrast, a call like `seq.declare_channel("ch1", "rydberg_global"` would have `args={"ch1", "rydberg_global")` and `kwargs={}`. 
In your code, you are assuming that both arguments are given as args, but this is not necessarily true and it will fail if that is not the case.

Reply:
Very clear.

Reply:
So If I well  understand your code I'll get a value error if `name` is stored in `kwargs` and not in `args` ?

Reply:
Not `name`, `channel_id`